### ISSUE TITLE
The war against manual id references

### ISSUE THREAD

**Issue by Zironic** [Arx-Game/arxcode#71]
I firmly believe there's almost no situation where having the user manually enter a numeric ID is the most user friendly option or neccesary.

In almost all situations the game should very accurately be able to take a name and match it to an ID after filtering out the valid options and in the cases where there is more then one match it can let the user choose between them

**Comment by TehomCD** [comment]
Yeah, I'd prefer that. I added a method for searching by name or ID or raising `CommandError` in cases where it's not found relatively recently to the ArxCommandMixin, so it'd be a matter of going back and using that in older commands.
